lib/ecto/query/builder.ex

```diff
@@ -266,6 +266,33 @@ defmodule Ecto.Query.Builder do
     {expr, params_acc}
   end
 
+  @over_agg %{
+    row_number: [:integer],
+    rank: [:integer],
+    dense_rank: [:integer],
+    percent_rank: [:float],
+    cume_dist: [:float],
+    ntile: [:integer, :integer], # TODO: what should we do with functions which require exactly integer type?
```

Comment:
Hi @josevalim! I have some questions here. 
The first question is about `integer` and `bigint`. In some cases, they are interchangeable but if function requires `integer` we can't just cast a value to `bigint`.

Reply:
@Anber we don't make a distinction from this side of things, so just `:integer` should be enough.

Reply:
We don't, but queries like `Schema |> select([r], nth_value(r.x, "42") |> over)` throw a error `function nth_value(integer, bigint) does not exist`.

Reply:
Right. You need to cast or pass a proper value. There is not much we can do here because Postgres is thinking that it is expecting a bigint.

Reply:
I agree that we can't do anything here right now, but maybe there is a possibility to add more strict type system in the future?